blinker/base.py

```diff
@@ -396,10 +396,7 @@ def signal(self, name, doc=None):
         Repeated calls to this function will return the same signal object.
 
         """
-        try:
-            return self[name]
-        except KeyError:
-            return self.setdefault(name, NamedSignal(name, doc))
+        return self.setdefault(name, NamedSignal(name, doc))
```

Comment:
yah, you're perfectly right . Anyway, does it really matter a lot when we just create several objects and throw them away  if this can make the code more graceful? This seems kind of subjective:)